js-scripts/generateRedemptionRights/index.ts

```diff
@@ -0,0 +1,352 @@
+import * as fs from "fs";
+import * as dotenv from "dotenv";
+import {
+  keccak256,
+  encodePacked,
+  createPublicClient,
+  http,
+  parseAbiItem,
+  Address,
+  zeroAddress,
+  PublicClient,
+  getContract,
+  erc20Abi,
+} from "viem";
+import { mainnet } from "viem/chains";
+import { MerkleTree } from "merkletreejs";
+
+// Types
+type PointsBalance = Map<`0x${string}`, bigint>;
+type RedemptionRightsMap = Map<Address, PointsBalance>;
+
+interface AlphaDistributionData {
+  pTokens: {
+    [address: Address]: {
+      [pointsId: `0x${string}`]: {
+        accumulatingPoints: string;
+      };
+    };
+  };
+}
+
+interface MerklizedData {
+  root: `0x${string}`;
+  redemptionRights: {
+    [address: Address]: {
+      [pointsId: `0x${string}`]: {
+        amount: string;
+        proof: `0x${string}`[];
+      };
+    };
+  };
+  pTokens: {
+    [address: Address]: {
+      [pointsId: `0x${string}`]: {
+        amount: string;
+        proof: `0x${string}`[];
+      };
+    };
+  };
+}
+
+interface PTokenSnapshot {
+  address: Address;
+  blockNumber: string;
+  balances: {
+    [address: string]: string;
+  };
+}
+
+// Overrides
+const UNI_POOL_OVERRIDES = {
+  "0x597a1b0515bbeEe6796B89a6f403c3fD41BB626C": {
+    "0x24C694d193B19119bcDea9D40a3b0bfaFb281E6D": 6487631537430741114,
+    "0x44Cb2d713BDa3858001f038645fD05E23E5DE03D": 27597767454066598826095,
+  },
```

Comment:
total amount looks like it adds up to the total amount in the LP pool. How did you divvy it up?

Reply:
Ahh, tbh I just impersonated the various addresses that had added to the uni pool at some point to see what the uni FE said they were able to withdraw

Reply:
nice! makes sense, was just curious